src/query_graph/field_set.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::{FederationError, MultipleFederationErrors, SingleFederationError};
+use crate::schema::position::CompositeTypeDefinitionPosition;
+use apollo_compiler::executable::{FieldSet, SelectionSet};
+use apollo_compiler::{NodeStr, Schema};
+
+// TODO: In the JS codebase, this optionally runs an additional validation to forbid aliases, and
+// has some error-rewriting to help give the user better hints around non-existent fields.
+pub(super) fn parse_field_set(
+    schema: &Schema,
+    parent_composite_type_definition_position: CompositeTypeDefinitionPosition,
```

Comment:
since we are just using the type name should we just accept the name?

Reply:
The original intention behind it was to force people to `try_into()` if they had some wider `*DefinitionPosition`, but arguably the function would be more portable if it used e.g. `NamedType` instead (and the callsites can still just call `try_into()` themselves). Pushed up a commit for the changes.